includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -142,6 +149,141 @@ function get_columns() {
 
 		return $this->columns;
 	}
+
+	/**
+	 * Get total average progress for all the courses.
+	 *
+	 * @since  4.3.0
+	 * @access private
+	 *
+	 * @return int|false average_progress total average progress value and false if not applicable.
+	 */
+	private function get_total_average_progress_for_courses_report() {
+		// Get all courses ids.
+		$courses_ids = $this->get_all_courses_ids();
+
+		if ( ! is_array( $courses_ids ) || empty( $courses_ids ) ) {
+			return false;
+		}
+		// Calculate total average progress for all the courses.
+		$average_course_progress_total = $this->get_total_average_progress_for_all_the_courses( $courses_ids );
+
+		// If total value for all the courses is zero, return N/A.
+		if ( ! $average_course_progress_total ) {
+			return false;
+		}
+
+		// Divide the value of all progresses combined with the number of courses to get average value.
+		if ( is_array( $courses_ids ) ) {
+			$average_course_progress_total = $average_course_progress_total / count( $courses_ids );
+		}
+
+		return ceil( $average_course_progress_total );
+	}
+
+	/**
+	 * Get students grouped by courses.
+	 *
+	 * @since  4.3.0
+	 *
+	 * @param array $courses_ids Array of courses.
+	 * @return object[][] $students_grouped_by_courses Students array grouped by courses.
+	 */
+	private function get_students_by_courses( $courses_ids ) {
+		// Get all students for courses.
+		$students = Sensei()->course->get_students_grouped_by_courses( $courses_ids );
+
+		// Get students grouped by courses.
+		$students_by_courses = array();
+		foreach ( $students as $student ) {
+			$students_by_courses[ $student->course_id ][] = $student->student_id;
+		}
+		return $students_by_courses;
+	}
+
+	/**
+	 * Get lessons grouped by courses.
+	 *
+	 * @since  4.3.0
+	 *
+	 * @param array $courses_ids Array of courses.
+	 * @return object[][] $lessons_grouped_by_courses Lessons array grouped by courses.
+	 */
+	private function get_lessons_by_courses( $courses_ids ) {
+		// Get Lessons by courses.
+		$lessons = Sensei()->course->get_lessons_grouped_by_courses( $courses_ids );
+
+		// Get Lessons grouped by courses.
+		$lessons_by_courses = array();
+		foreach ( $lessons as $lesson ) {
+			$lessons_by_courses[ $lesson->course_id ][] = $lesson->lesson_id;
+		}
+		return $lessons_by_courses;
+	}
+
+	/**
+	 * Get total average progress for all the courses.
+	 *
+	 * @since  4.3.0
+	 * @access private
+	 *
+	 * @param array $courses_ids List of courses ids.
+	 * @return string average_progress total average progress value and N/A if not applicable.
+	 */
+	private function get_total_average_progress_for_all_the_courses( $courses_ids ) {
+		$average_course_progress_total = 0;
+		if ( ! is_array( $courses_ids ) || empty( $courses_ids ) ) {
```

Comment:
No need to check if this is an array if you have the above-mentioned "contract".